lib/frizz/remote.rb

```diff
@@ -13,14 +13,40 @@ def files
       @files ||= objects.reject { |o| ignore?(o) }
     end
 
+    def file_local_path(remote_file_key)
+      # since all s3 uploads don't have .gz extension, we need to determine
+      # if the local_path should have the .gz extension when comparing local and remote files
+      head_object = client.head_object({
+        bucket: bucket_name,
+        key: remote_file_key
+      })
+
+      if head_object.content_encoding == 'gzip'
+        "#{remote_file_key}.gz"
+      else
+        remote_file_key
+      end
+    end
```

Comment:
Yeah much of this stuff (above and below in this file) could be moved out into an object with the responsibility of local<->remote mapping which would act as a collaborator to `Local` and `Remote` (and perhaps `Ignorance` as well — although I'd look for a way to reduce its knowledge of this mapping object)